doc/user/content/transform-data/optimization.md

```diff
@@ -320,29 +320,29 @@ FROM (
+However, the estimate is based only on data that is already present in the system. So taking into account how much this largest number could expand is critical to avoid issues with update latency after tuning the query hint.
 
-For our example, let's suppose that we determined the largest number of courses per teacher to be `1000`. Then, the original definition of `max_course_id_per_teacher` can be revised to include the `EXPECTED GROUP SIZE` query hint as follows:
+For our example, let's suppose that we determined the largest number of courses per teacher to be `1000`. Then, the original definition of `max_course_id_per_teacher` can be revised to include the `AGGREGATE INPUT GROUP SIZE` query hint as follows:
 
 ```sql
 CREATE MATERIALIZED VIEW max_course_id_per_teacher AS
 SELECT teacher_id, MAX(course_id)
 FROM sections
 GROUP BY teacher_id
-OPTIONS (EXPECTED GROUP SIZE = 1000)
+OPTIONS (AGGREGATE INPUT GROUP SIZE = 1000)
 ```
 
-The same hint can be provided in [Top K] query patterns specified by `DISTINCT ON` or `LATERAL`. As examples, consider that we wish not to compute the maximum `course_id`, but rather the `id` of the section of this top course. This computation can be incrementally maintained by the following materialized view:
+The other two hints can be provided in [Top K] query patterns specified by `DISTINCT ON` or `LIMIT`. As examples, consider that we wish not to compute the maximum `course_id`, but rather the `id` of the section of this top course. This computation can be incrementally maintained by the following materialized view:
 
 ```sql
 CREATE MATERIALIZED VIEW section_of_top_course_per_teacher AS
 SELECT DISTINCT ON(teacher_id) teacher_id, id AS section_id
 FROM sections
-OPTIONS (EXPECTED GROUP SIZE = 1000)
+OPTIONS (DISTINCT ON INPUT GROUP SIZE = 1000)
 ORDER BY teacher_id ASC, course_id DESC;
 ```
 
-In the above examples, we see that the `EXPECTED GROUP SIZE` hint is always positioned in an `OPTIONS` clause after a `GROUP BY` clause, but before an `ORDER BY`, as captured by the [`SELECT` syntax]. However, in the case of Top K using a `LATERAL` subquery and `LIMIT`, the hint is specified in the subquery. For instance, the following materialized view illustrates how to incrementally maintain the top-3 section `id`s ranked by `course_id` for each teacher:
+In the above examples, we see that the query hints are always positioned in an `OPTIONS` clause after a `GROUP BY` clause, but before an `ORDER BY`, as captured by the [`SELECT` syntax]. However, in the case of Top K using a `LATERAL` subquery and `LIMIT`, it is important to note that the hint is specified in the subquery. For instance, the following materialized view illustrates how to incrementally maintain the top-3 section `id`s ranked by `course_id` for each teacher:
```

Comment:
You mention `LATERAL` here, but removed it above; do these hints apply to statements using `LIMIT` **or** `LATERAL` statements, or is `LATERAL` no longer applicable?